src/runners/testngRunner/TestNGRunnerResultAnalyzer.ts

```diff
@@ -40,7 +41,7 @@ export class TestNGRunnerResultAnalyzer implements IRunnerResultAnalyzer {
     }
 
     public analyzeData(data: string): void {
-        const lines: string[] = data.split(/\r?\n/);
+        const lines: string[] = this.unescape(data).split(/\r?\n/);
```

Comment:
I found the code in `analyzeData()` has some bugs, which I changed it to:

```js
public analyzeData(data: string): void {
    let match: RegExpExecArray | null;
    // tslint:disable-next-line: no-conditional-assignment
    while ((match = this.regex.exec(data)) !== null) {
        try {
           this.processData(match[1]);
        } catch (error) {
            this.testContext.testRun.appendOutput(`[ERROR] Failed to parse output data: ${match[1]}\n`);
        }
    }
}
```

Meanwhile, add the `g` flag to the regexp @ line 16:

```js
private readonly regex: RegExp = /@@<TestRunner-({[\s\S]*?})-TestRunner>/g;
```

Reply:
Ah, I see why it's not working. Because here `.unescape(data)` is called before splitting to lines. Is there any reason to do this?